examples/api/lib/material/menu_bar/create_material_menu.0.dart

```diff
@@ -0,0 +1,238 @@
+  set backgroundColor(Color value) {
+    if (_backgroundColor != value) {
+      setState(() {
+        _backgroundColor = value;
+      });
+    }
+  }
+
+  void _activate(MenuSelection selection) {
+    setState(() {
+      _lastSelection = selection;
+    });
+
+    switch (selection) {
+      case MenuSelection.about:
+        showAboutDialog(
+          context: context,
+          applicationName: 'MenuBar Sample',
+          applicationVersion: '1.0.0',
+        );
+        break;
+      case MenuSelection.showMessage:
+        showingMessage = !showingMessage;
+        break;
+      case MenuSelection.resetMessage:
+      case MenuSelection.hideMessage:
+        showingMessage = false;
+        break;
+      case MenuSelection.colorMenu:
+        break;
+      case MenuSelection.colorRed:
+        backgroundColor = Colors.red;
+        break;
+      case MenuSelection.colorGreen:
+        backgroundColor = Colors.green;
+        break;
+      case MenuSelection.colorBlue:
+        backgroundColor = Colors.blue;
+        break;
+    }
+  }
+
+  void _updateMenu() {
+    _menuHandle = createMaterialMenu(
+      buttonFocusNode: _buttonFocusNode,
+      controller: _controller,
+      children: <Widget>[
+        MenuItemButton(
+          child: Text(MenuSelection.about.label),
+          onPressed: () => _activate(MenuSelection.about),
+        ),
+        // Toggles the message.
+        MenuItemButton(
+          onPressed: () => _activate(MenuSelection.showMessage),
+          shortcut: MenuSelection.showMessage.shortcut,
+          child: Text(
+            showingMessage ? MenuSelection.hideMessage.label : MenuSelection.showMessage.label,
+          ),
+        ),
+        // Hides the message, but is only enabled if the message isn't already hidden.
+        MenuItemButton(
+          onPressed: showingMessage ? () => _activate(MenuSelection.resetMessage) : null,
+          shortcut: MenuSelection.resetMessage.shortcut,
+          child: Text(MenuSelection.resetMessage.label),
+        ),
+        MenuButton(
+          menuChildren: <Widget>[
+            MenuItemButton(
+              onPressed: () => _activate(MenuSelection.colorRed),
+              shortcut: MenuSelection.colorRed.shortcut,
+              child: Text(MenuSelection.colorRed.label),
+            ),
+            MenuItemButton(
+              onPressed: () => _activate(MenuSelection.colorGreen),
+              shortcut: MenuSelection.colorGreen.shortcut,
+              child: Text(MenuSelection.colorGreen.label),
+            ),
+            MenuItemButton(
+              onPressed: () => _activate(MenuSelection.colorBlue),
+              shortcut: MenuSelection.colorBlue.shortcut,
+              child: Text(MenuSelection.colorBlue.label),
+            ),
+          ],
+          child: const Text('Background Color'),
+        ),
+      ],
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _updateMenu();
```

Comment:
Actually, that does make it more complex: If the state object is a `ChangeNotifier`, then it can't notify until the next frame, since then it could be notifying a widget that was drawn in the overlay.

Reply:
Hm, I see. However, in the current implementation, if `MyCascadingMenu` (the widget that owns the menu handle) rebuilds for unrelated reasons (maybe because one of its dependencies changed, etc) that would close the menu because we dispose the handle on build, no? To the user this would look like an annoying glitch when the menu closes for reasons not obvious to them? In theory, the `build` method could be called every frame (that it isn't is just an optimization in the framework)...

Also: If a widget owning a menu happens to get rebuild frequently for reasons unrelated to the menu, we are doing a bunch of extra work in every build which isn't necessary if the menu isn't shown. Since menus are rather temporary, maybe they should only do this work if the menu is actually shown?

Reply:
Perhaps I should just modify the example to not show updated state in the menu in response to a state change here?

Reply:
Since this is the introductory example, that would be fair to simplify it like that.

Still leaves the question of how you would do this in a real app if you need that? Maybe the answer is you'd just use package:provider or whatever to inject the data you want into the menu and package:provifder would take care of rebuilding locally?

Reply:
Yeah, I think that's basically the answer.  I mean, I could do it here too using a `ChangeNotifier` and scheduling the update post-frame, but that seems like overkill for an example.

In the future, when we have `OverlayPortal`, I think this will get a lot easier, since it would just listen to a change notifier as normal, and wouldn't need to delay a frame.

Reply:
Also, this is using `createMaterialMenu`, which is a slightly lower level API. If they use `MenuBar`, then the updating is taken care of for them.